src/test_workflow/integ_test/service_opensearch_dashboards.py

```diff
@@ -38,6 +38,7 @@ def __init__(
         super().__init__(work_dir, version, distribution, security_enabled, additional_config, dependency_installer)
         self.dist = Distributions.get_distribution("opensearch-dashboards", distribution, version, work_dir)
         self.install_dir = self.dist.install_dir
+        self.password = 'myStrongPassword123!' if float('.'.join(self.version.split('.')[:2])) >= 2.12 else 'admin'
```

Comment:
This PR should address this failure: https://build.ci.opensearch.org/blue/organizations/jenkins/integ-test-opensearch-dashboards/detail/integ-test-opensearch-dashboards/5025/pipeline, however I am not sure how the OS binary is pulled - is it given `myStrongPassword123!` as the password somehow?

Reply:
@gaiksaya done. Question still stands though, I am confused about the overall setup of: https://build.ci.opensearch.org/blue/organizations/jenkins/integ-test-opensearch-dashboards/detail/integ-test-opensearch-dashboards/5025/pipeline/, would this change be sufficient to pass the integ test? Would the backend be spun up with the strong password as expected if version >= 2.12.0?

Reply:
The integration test set up can be found here: https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/integ_test/ for both OS and OSD. Looks like needs update here too: https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/integ_test/service_opensearch_dashboards.py#L89

Maybe try to parse this directory to check for anything admin:admin?
If it feels there is a lot of if/else going on each module, maybe extract the logic into a common library (utils?) (something like get_password(version)) that can be used universally. Example being this [system](https://github.com/opensearch-project/opensearch-build/tree/main/src/system) module